docs/theory.rst

```diff
@@ -43,29 +43,244 @@ By collecting :math:`x_n` into a complex number array (called :cpp:member:`Qrack
+
+.. math::
+    :label: x_3bit
+
+    (X \otimes I \otimes I) \times M
+
+.. math::
+    :label: x_3bit_2
+
+    (\begin{bmatrix}
+      0 & 1 \\\
+      1 & 0
+    \end{bmatrix}
+    \otimes
+    \begin{bmatrix}
+      1 & 0 \\\
+      0 & 1
+    \end{bmatrix}
+    \otimes
+    \begin{bmatrix}
+      1 & 0 \\\
+      0 & 1
+    \end{bmatrix}) \times
+    \begin{bmatrix}
+      x_{000} \\
+      x_{001} \\
+      x_{010} \\
+      x_{011} \\
+      x_{100} \\
+      x_{101} \\
+      x_{110} \\
+      x_{111}
+    \end{bmatrix}
+
+.. math::
+    :label: x_3bit_3
+
+    \begin{bmatrix}
+      0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 \\
+      1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 \\
+      0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 \\
+      0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 \\
+      0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 \\
+      0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 \\
+      0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 \\
+      0 & 0 & 0 & 0 & 0 & 0 & 1 & 0
+    \end{bmatrix}
+    \times
+    \begin{bmatrix}
+      x_{000} \\
+      x_{001} \\
+      x_{010} \\
+      x_{011} \\
+      x_{100} \\
+      x_{101} \\
+      x_{110} \\
+      x_{111}
+    \end{bmatrix}
 
-    qReg->X(8, 8);
+.. math::
+  :label: x_3bit_final
+
+    (X \otimes I \otimes I) \times 
+    \begin{bmatrix}
+      x_{000} \\
+      x_{001} \\
+      x_{010} \\
+      x_{011} \\
+      x_{100} \\
+      x_{101} \\
+      x_{110} \\
+      x_{111}
+    \end{bmatrix}
+    = 
+    \begin{bmatrix}
+      x_{001} \\
+      x_{000} \\
+      x_{011} \\
+      x_{010} \\
+      x_{101} \\
+      x_{100} \\
+      x_{111} \\
+      x_{110}
+    \end{bmatrix}
+
+The equation :eq:`x_3bit` inverts the probability of the first bit out of three, but leave the second and third bits alone.  Using the identity matrix :math:`I` preserves the probabilities of the :math:`x_{0nn}` and :math:`x_{1nn}` positions.  The expanded matrix in :eq:`x_3bit_3` now has the proper dimensionality to be multiplied directly against the probability matrix.
+
+.. note:: It's important to remember here that, unlike a classical :math:`NOT` which directly inverts a bit, the :math:`X` gate swaps the *probabilities* for the states where the qubit is 1 with the probabilities where the qubit is 0.  So while :math:`x_{000}` and :math:`x_{100}` have particular complex number values, the position in the matrix :math:`M[0]` will always correspond to the probability :math:`x_0` in :eq:`3qb_prob`.  If the value of :math:`M[0]` is :math:`x_{100}`, then the probability of the system, on measurement, resulting in :math:`\rvert000\rangle` is equal to the probability that the system, prior to the :math:`X` gate, would have resulted in :math:`\rvert100\rangle`.  See `Quantum Logic Gates <https://en.wikipedia.org/wiki/Quantum_logic_gate#Circuit_composition_and_entangled_states>`_ for more information.
```

Comment:
Change wording throughout to "amplitude" rather than "probability," where appropriate. The amplitudes are swapped.